File: xiaoming/1025/FindMinimumInRotatedSortedArray.java

public class FindMinimumInRotatedSortedArray {
	public int findMin(int[] nums) {
        if (nums == null || nums.length == 0) return -1;
        int start = 0;
        int end = nums.length -1;
        int mid = 0;
        while(start <= end) {
            mid = (start + end) / 2;
            if (nums[mid] > nums[nums.length - 1]) start = mid + 1;
            else end = mid -1;
        }
        return nums[end + 1];
    }
}

File: Sen/Find_Minimum_in_Rotated_Sorted_Array.py
class Solution(object):
	def findMin(self, nums):
		low, high = 0, len(nums) - 1
		while low < high:
			if nums[low] < nums[high]:
				return nums[low]
			mid = low + (high - low) / 2
			# if nums[mid] < nums[high]:
			# 	high = mid
			# else:
			# 	low = mid + 1
			if nums[mid] >= nums[low]:
				low = mid + 1
				print mid, low, high
			else:
				high = mid
				print mid, low, high
		return nums[low]


# nums = [4, 5, 6, 7, 1, 2, 3]
# nums = [7, 1, 2, 3]
nums = [7, 1]
i = Solution()
print i.findMin(nums)
